src/notebooklm/cli/helpers.py

```
@@ -90,19 +90,80 @@ async def import_with_retry(
     started_at = time.monotonic()
     delay = initial_delay
     attempt = 1
 
+    requested_urls = {url for s in sources if isinstance((url := s.get("url")), str) and url}
+
+    # Snapshot baseline source URLs so we can detect server-side success on
+    # timeout by computing the delta after each failed import RPC.
+    try:
+        baseline = await client.sources.list(notebook_id)
+        baseline_urls: set[str] | None = {src.url for src in baseline if src.url}
+    except Exception as snapshot_exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)